src/manifest.c

```diff
@@ -554,6 +564,13 @@ void subtract_manifests(struct manifest *m1, struct manifest *m2)
 		}
 		list2 = g_list_next(list2);
 	}
+
+	/* now recurse into the included manifests and subtract those as well */
+	while (includes) {
+		mi = includes->data;
+		includes = g_list_next(includes);
+		subtract_manifests(m1, mi, includes);
```

Comment:
Last thing I've noticed so far: the subtraction for the "openstack-all-in-one" bundle seems to be in an infinite recursion loop. Since the number of subtractions happening is quite large and includes lots of duplicates, I haven't determined why this is happening yet...